code/game/machinery/fabricators/modular_fabricator.dm

```diff
@@ -493,13 +493,13 @@
 	use_power(power)
 	materials.use_materials(materials_used)
 	if(is_stack)
-		var/obj/item/stack/N = new being_built.build_path(null, multiplier)
-		N.forceMove(A)
+		var/obj/item/stack/N = new being_built.build_path(A, multiplier)
+		N.forceMove(A) //Forcemove to the same spot to trigger ZFall
 		N.update_icon()
 	else
 		for(var/i in 1 to multiplier)
-			var/obj/item/new_item = new being_built.build_path(null)
-			new_item.forceMove(A)
+			var/obj/item/new_item = new being_built.build_path(A)
+			new_item.forceMove(A) //Forcemove to the same spot to trigger ZFall
```

Comment:
what's this change?

Reply:
The last line in the CL.

 #10855 made objects from fabricators fall if outputted to openspace by putting them in nullspace and then moving them to the release turf.

This resulted in a different bug. When a positronic was created, you'd get "A Positronic brain has been created in null!" which was not very useful. It's because the positronic was created so it ran the init function in nullspace, then got moved to the release turf.

This fixes it (after the commit I just pushed because I'm an idiot).

What it does is fabricate the object on the location of the fabricator, and then forcemove it to the release turf, which then triggers ZFall. That way the init function gets run in the same area as the posi will end up (or at least nearby).

There's an edge case here. If the output turf is the fabricator, and there is openspace below the fabricator (hovering), the output will not fall, because the destination turf is the same so forcemove returns early. I figure this is fine, as the fabricator is "supporting" the object, and really shouldn't be in that location anyways because it's hovering.